spoon/template/compiler.php

```diff
@@ -775,28 +775,28 @@ protected function parseOptions($content)
 						$replace[] = '<?php
 						if(
 							(is_object(' . $baseVariable . ') && ' . $objectVariable . ' && ' . $objectVariable . ' != \'\' && ' . $objectVariable . ' !== false)
-							|| (is_array(' . $baseVariable . ') && isset(' . $variable . ') && count(' . $variable . ') != 0 && ' . $variable . ' != \'\' && ' . $variable . ' !== false))
+							|| (is_array(' . $baseVariable . ') && isset(' . $variable . ') && empty(' . $variable . ') === false && ' . $variable . ' != \'\' && ' . $variable . ' !== false))
 						{
 							?>';
 						$replace[] = '<?php } ?>';
 
 						// inverse option
 						$replace[] = '<?php if(
 							(is_object(' . $baseVariable . ') && (!' . $objectVariable . ' || ' . $objectVariable . ' == \'\' || ' . $objectVariable . ' === false))
-							|| (is_array(' . $baseVariable . ') && (!isset(' . $variable . ') || count(' . $variable . ') == 0 || ' . $variable . ' == \'\' || ' . $variable . ' === false))): ?>';
+							|| (is_array(' . $baseVariable . ') && (!isset(' . $variable . ') || empty(' . $variable . ') === true || ' . $variable . ' == \'\' || ' . $variable . ' === false))): ?>';
 						$replace[] = '<?php endif; ?>';
 					}
 					else
 					{
 						// replace with
 						$replace[] = '<?php
-						if(isset(' . $variable . ') && count(' . $variable . ') != 0 && ' . $variable . ' != \'\' && ' . $variable . ' !== false)
+						if(isset(' . $variable . ') && empty(' . $variable . ') === false && ' . $variable . ' != \'\' && ' . $variable . ' !== false)
 						{
 							?>';
 						$replace[] = '<?php } ?>';
 
 						// inverse option
-						$replace[] = '<?php if(!isset(' . $variable . ') || count(' . $variable . ') == 0 || ' . $variable . ' == \'\' || ' . $variable . ' === false): ?>';
```

Comment:
doen't empty also check for empty strings and false? in that case the checks after this can be removed

Reply:
![schermafbeelding 2018-03-28 om 10 18 13](https://user-images.githubusercontent.com/16104135/38017116-5faef690-3271-11e8-8e5c-db5160b03fe8.png)
Found this nice diagram, these can be omitted indeed.

Reply:
I think in general if we refactor every line to working with isset and empty, we got the whole list covered. So maybe best to do this on every line.